### DISCUSSION TITLE
[Feature Request] Temperature slider, other parameters

### DISCUSSION THREAD

**Discussion by Kristovich** [danny-avila/LibreChat#77] [Feature Requests & Suggestions]
OpenAi exposes a few parameters for chatgpt, but the only parameter that I've found most useful is temperature. By setting temperature, I can get it to follow instructions a bit better and it seems less likely to make stuff up.

One project that I think exposes these parameters well is [FreeChatGPT](https://github.com/ztjhz/FreeChatGPT). They have a bar on the top of the page with the current parameters for the session. 
![image](https://user-images.githubusercontent.com/22111725/225695903-28e88e39-308f-4abd-9836-9a0cb6a4854d.png)

When clicked, it opens up this menu of sliders.
![image](https://user-images.githubusercontent.com/22111725/225696438-f6d9bf5e-2cd7-4b05-b6a4-29c24bc06c6a.png)

That can be freely changed during the conversation. 

I think setting custom parameters like this would be a useful addition, especially for the "CustomGPT" option since its persistent. I find myself stetting temperature down to 0 a lot when trying to get chatgpt to explain something to me.

**Comment by danny-avila** [comment]
Great idea, I'll add this to the roadmap. I have a lot of time this coming weekend and hopefully can get to it in my sprint then

**Comment by wtlyu** [comment]
![image-4](https://user-images.githubusercontent.com/8267138/229333301-9236cb1c-9fce-48d2-b8f5-dfee20a35d3b.png)
![image-3](https://user-images.githubusercontent.com/8267138/229333306-3347baf5-feb8-4628-84c6-94d441b1b77f.png)
![image-2](https://user-images.githubusercontent.com/8267138/229333310-dad4b066-e7af-4863-8b3b-2a2ecf3df935.png)
 working in progress.

**Comment by alfredo-f** [comment]
What about the maximum length?

![image](https://user-images.githubusercontent.com/22865959/235288802-d3303459-af1c-4fc0-8b07-87fe8b3678bd.png)
